ravenframework/Models/Model.py

```diff
@@ -298,19 +316,25 @@ def _replaceVariablesNamesWithAliasSystem(self, sampledVars, aliasType='input',
       listAliasType = [aliasType]
     originalVariables = copy.deepcopy(sampledVars)
     for aliasTyp in listAliasType:
-      for varFramework,varModel in self.alias[aliasTyp].items():
-        whichVar =  varModel if fromModelToFramework else varFramework
-        notFound = 2**62
-        if type(originalVariables).__name__ != 'list':
-          found = sampledVars.pop(whichVar,[notFound])
-          if not np.array_equal(np.asarray(found), [notFound]):
-            if fromModelToFramework:
-              sampledVars[varFramework] = originalVariables[varModel]
-            else:
-              sampledVars[varModel]     = originalVariables[varFramework]
+      for varFramework, varModel in self.alias[aliasTyp].items():
+        oldName = varModel if fromModelToFramework else varFramework
+        newName = varFramework if fromModelToFramework else varModel
+        if isinstance(originalVariables, dict):
+          # replace the old name with the new one, if present
+          if oldName in sampledVars:
+            value = sampledVars.pop(oldName)
+            sampledVars[newName] = value
+        elif isinstance(sampledVars, list):
```

Comment:
I realize `originalVariables` is a deepcopy of `sampledVars`, but it would be nice if this set of `if isinstance` checked on the same variable instead of the two different ones.